pkg/management/delete_user_defined_alert_rule_by_id.go

```
@@ -119,14 +133,39 @@ func (c *client) deleteUserAlertRuleById(ctx context.Context, namespace, name, a
+		return nil
+	}
+	return c.deleteAssociatedARC(ctx, arcNamespace, name, alertRuleId)
+}
+
+// deleteAssociatedARC removes the AlertRelabelConfig associated with an alert rule, if it exists.
+// This is best-effort: if the ARC does not exist or is GitOps-managed, it is silently skipped.
+func (c *client) deleteAssociatedARC(ctx context.Context, namespace, prName, alertRuleId string) error {
+	arcName := k8s.GetAlertRelabelConfigName(prName, alertRuleId)
+	arc, found, err := c.k8sClient.AlertRelabelConfigs().Get(ctx, namespace, arcName)
+	if err != nil || !found {
+		return nil
```

golangci-lint (2.12.2)

[error] error is not nil (line 161) but it returns nil

(nilerr)